src/figuresClasses.ts

```diff
@@ -1,19 +1,76 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public aSide: number,
+    public bSide: number,
+    public cSide: number,
+  ) {
+    if (aSide <= 0 || bSide <= 0 || cSide <= 0) {
+      throw new Error('One of the figure\'s sides less or 0.');
+    }
+
+    if (Math.max(aSide, bSide, cSide) >= (aSide + bSide + cSide)
```

Comment:
better will be `if (aSide + bSide <= cSide || aSide + cSide <= bSide || bSide + cSide <= aSide)`.